scripts/Diffraction/isis_powder/routines/focus.py

```diff
@@ -48,22 +48,47 @@ def _focus_one_ws(input_workspace, run_number, instrument, perform_vanadium_norm
         summed_empty = common.generate_summed_runs(empty_sample_ws_string=run_details.sample_empty,
                                                    instrument=instrument,
                                                    scale_factor=instrument._inst_settings.sample_empty_scale)
-    if summed_empty is not None:
+
+    # Correct for absorption / multiple scattering if required
+    if absorb and instrument._inst_settings.absorb_method == 'PaalmanPings' and summed_empty:
+        events_per_point_string = instrument._inst_settings.paalman_pings_events_per_point
+        if events_per_point_string:
+            events_per_point = int(events_per_point_string)
+        else:
+            events_per_point = 1000
+        expected_filename = "PaalmanPingsApplied_" + instrument.get_instrument_prefix() + run_number \
+                            + "_" + instrument._get_input_batching_mode() \
+                            + f"_{events_per_point}eventsperpoint" + ".nxs"
+        save_directory = instrument._inst_settings.output_dir
+        if expected_filename in os.listdir(save_directory):
+            input_workspace = mantid.LoadNexus(os.path.join(save_directory, expected_filename))
+        else:
+            mantid.SetSample(input_workspace, Geometry=sample_details.generate_sample_geometry(),
+                             Material=sample_details.generate_sample_material(),
+                             ContainerGeometry=sample_details.generate_container_geometry(),
+                             ContainerMaterial=sample_details.generate_container_material())
+
+            corrections = mantid.PaalmanPingsMonteCarloAbsorption(InputWorkspace=input_workspace, Shape='Preset',
+                                                                  EventsPerPoint=events_per_point)
+
+            input_workspace = mantid.ApplyPaalmanPingsCorrection(SampleWorkspace=input_workspace,
+                                                                 CorrectionsWorkspace=corrections,
+                                                                 CanWorkspace=summed_empty)
+            mantid.SaveNexus(input_workspace, os.path.join(save_directory, expected_filename))
```

Comment:
This saved file of the applied paalmanpings workspace has no knowledge of the sample/container shape/material so may be reloaded ignoring any changed input values. So loading a saved file should probably only happen if the inputs to SetSample() are identical